src/Network/Http/Auth/Oauth.php

```diff
@@ -54,10 +49,33 @@ public function authentication(Request $request, array $credentials)
         $credentials['method'] = strtoupper($credentials['method']);
         switch ($credentials['method']) {
             case 'HMAC-SHA1':
+                $hasKeys = isset(
+                    $credentials['consumerSecret'],
+                    $credentials['token'],
+                    $credentials['tokenSecret']
+                );
+                if (!$hasKeys) {
+                    return;
+                }
                 $value = $this->_hmacSha1($request, $credentials);
                 break;
 
+            case 'RSA-SHA1':
+                if (!isset($credentials['privateKeyFile'])) {
+                    return;
```

Comment:
Sry, but you confused me a bit now. :smile: You mean, we can do it in another PR? So we can finish this PR now? ^^